.github/workflows/lint.yml

```diff
@@ -35,13 +35,12 @@ jobs:
       - uses: actions/checkout@v3.1.0
         with:
           persist-credentials: false
-          fetch-depth: 2
+          fetch-depth: 0
```

Comment:
Are you sure about this?
How does it work out the changed files with only one commit?

Reply:
Here's the explanation why we need to use `fetch-depth: 0` for our use case. Is not a great UX, but this solves our issue (it compares the whole PR against `main` instead of doing it with the previous `sha`...which we prefer anyways) https://github.com/tj-actions/changed-files/issues/639#issuecomment-1256442729

Reply:
Thanks, can you add a comment or some documentation explaining that, and linking to the ticket?

It doesn't have to be in this PR. But I think it's worth doing because these errors have been confusing for a while.

Reply:
I've created an issue to do it in a different PR so we don't reset the tests here, as we need it merge. https://github.com/ZcashFoundation/zebra/issues/5373